pallets/subtensor/src/subnets/uids.rs

```diff
@@ -9,6 +9,22 @@ impl<T: Config> Pallet<T> {
         SubnetworkN::<T>::get(netuid)
     }
 
+    /// Sets value for the element at the given position if it exists.
+    pub fn set_element_at<N>(vec: &mut [N], position: usize, value: N) {
+        if let Some(element) = vec.get_mut(position) {
+            *element = value;
+        }
+    }
+
+    /// Resets the trust, emission, consensus, incentive, dividends of the neuron to default
+    pub fn clear_neuron(netuid: u16, neuron_uid: u16) {
```

Comment:
Can we optimise this by reducing the number of storage / writes , and number of function calls ? 

```rust
pub fn clear_neuron(netuid: u16, neuron_uid: u16) {
    let neuron_index: usize = neuron_uid.into();
    let default_value = 0;

    for storage in &mut [
        &mut Emission::<T>::get(netuid),
        &mut Trust::<T>::get(netuid),
        &mut Consensus::<T>::get(netuid),
        &mut Incentive::<T>::get(netuid),
        &mut Dividends::<T>::get(netuid),
    ] {
        set_element_at(storage, neuron_index, default_value);
    }
}
```

Reply:
that does not quite work as the vectors have different int types

Reply:
it's the same number of calls to the set_element_at. is there much difference between get vs mutate calls? aren't they compiled to similar things?

Reply:
anyway made the change

Reply:
reverted - that does not work

Reply:
Can we not just clear the netuid prefix? Or is that the same runtime?

Reply:
it does not modify the vector - &mut Emission::<T>::**get**(netuid) just gives back the value & StorageMap does not have a way to get a mut ref to the value